server/src/librelay/relay_server.go

```diff
@@ -331,8 +339,35 @@ func (relay *relayServer) awaitTransactionMined(tx *types.Transaction) (err erro
 	return nil
 }
 
-func (relay *relayServer) UnregisterRelay() error {
-	return relay.Unstake()
+func (relay *relayServer) RemoveRelay(ownerKey *ecdsa.PrivateKey) (err error) {
+	tx, err := relay.sendRemoveTransaction(ownerKey)
+	if err != nil {
+		return err
+	}
+	return relay.awaitTransactionMined(tx)
+}
+
+func (relay *relayServer) sendRemoveTransaction(ownerKey *ecdsa.PrivateKey) (tx *types.Transaction, err error) {
+	auth := bind.NewKeyedTransactor(ownerKey)
+	nonceMutex.Lock()
```

Comment:
here's an example why the code doesn't belong here: the nonce is of the OWNER, not of the RELAY. no need to lock or increment.
In fact, its strange that it even pass tests with it.